components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/DCRConfigurationMgtServiceImpl.java

```diff
@@ -44,6 +82,178 @@ public DCRConfiguration getDCRConfiguration() throws DCRMServerException {
     public void setDCRConfiguration(DCRConfiguration dcrConfiguration)
             throws DCRMServerException, DCRMClientException {
 
-        DCRConfigUtils.setDCRConfiguration(dcrConfiguration);
+        try {
+            validateMandateSSA(dcrConfiguration);
+            ResourceAdd resourceAdd = parseConfig(dcrConfiguration);
+            getConfigurationManager().replaceResource(DCR_CONFIG_RESOURCE_TYPE_NAME, resourceAdd);
+        } catch (ConfigurationManagementException e) {
+            throw handleServerException(ERROR_CODE_DCR_CONFIGURATION_RETRIEVE, e);
+        }
+    }
+
+    /**
+     * Configuration Management API returns a ConfigurationManagementException with the error code CONFIGM_00017 when
+     * resource is not found. This method wraps the original method and returns null if the resource is not found.
+     *
+     * @param resourceTypeName Resource type name.
+     * @param resourceName     Resource name.
+     * @return Retrieved resource from the configuration store. Returns {@code null} if the resource is not found.
+     * @throws ConfigurationManagementException exception
+     */
+    private static Resource getResource(String resourceTypeName, String resourceName)
```

Comment:
why this needs to be static? please check other methods also why it needs to be static?

Reply:
To have unit tests for those private methods. That way we can mockStatic the class and have unit tests.
Otherwise we need to make those methods public to have unit tests right?

Reply:
making static for unit tests is not a good development practise. We can make it a private and mock the 3rd party Class used inside the method. Please chcek